apps/api/src/interview-slot/interview-slot.controller.ts

```diff
@@ -10,8 +20,45 @@ export class InterviewSlotController {
   constructor(private readonly interviewSlotService: InterviewSlotService) {}
 
   @Get()
-  async getAll() {
-    return await this.interviewSlotService.getAll();
+  async getAll(
+    @Query('disciplines') disciplines: string,
+    @Query('interviewers') interviewers: string,
+  ) {
+    let interviewSlotsFilteredByDiscipline = [];
+    let interviewSlotsFilteredByInterviewer = [];
+    let allInterviewSlots = [];
+    if (disciplines) {
+      interviewSlotsFilteredByDiscipline =
+        await this.interviewSlotService.getByDisciplines(
+          disciplines ? disciplines.split(',') : [],
+        );
+    }
+    if (interviewers) {
+      interviewSlotsFilteredByInterviewer =
+        await this.interviewSlotService.getByInterviewers(
+          interviewers ? interviewers.split(',') : [],
+        );
+    }
+    if (!interviewers && !disciplines) {
+      allInterviewSlots = await this.interviewSlotService.getAll();
+    }
+    return [
+      ...allInterviewSlots,
+      ...interviewSlotsFilteredByDiscipline,
+      ...interviewSlotsFilteredByInterviewer,
+    ];
+  }
+
+  @Delete('/:id')
+  async deleteInterviewSlot(@Param('id') id: string) {
+    return await this.interviewSlotService.deleteInterviewSlot(id);
+  }
+
+  @Post()
+  async createInterviewSlot(@Body() interviewSlotDto: CreateInterviewSlotDto) {
```

Comment:
ovo bih isto rado da imamo botun SAVE u aplikaciji pa pošalje sve nove intervjue kao array, a ne 200 requestova pravit

Reply:
također, volio bih da su novi intervjui u tom slučaju prije nego se klikne SAVE uokvireni i da imamo negdje ispisano koliko je novih intervjua i kakvih

Reply:
ovako dodaš i ne znaš što se promijenilo